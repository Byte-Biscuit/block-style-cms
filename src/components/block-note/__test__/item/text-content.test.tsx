import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import FormattedText from "../../item/text-content";
import type { TextContent } from "../../meta";

describe("FormattedText", () => {
    it("renders text and applies style classes from styles object", () => {
        const item = {
            type: "text",
            text: "试文",
            styles: {
                bold: true,
                italic: true,
                underline: true,
                strike: false,
                textColor: "red",
                backgroundColor: "blue",
            },
        } as TextContent;

        //const { asFragment, container } = render(<FormattedText item={item} />);
        const { asFragment } = render(<FormattedText item={item} />);
        //screen.debug();
        //console.log(container.innerHTML);
        const el = screen.getByText("试文") as HTMLElement;
        expect(el).toBeTruthy();

        const className = el.className;
        // basic text style classes
        expect(className).toContain("font-bold");
        expect(className).toContain("italic");
        expect(className).toContain("underline");
        expect(className).not.toContain("line-through");

        // color mapping (from meta.twColorMap)
        // red -> text-red-600
        expect(className).toContain("text-red-600");
        // blue -> bg-sky-100 (background mapping)
        expect(className).toContain("bg-sky-100");

        expect(asFragment()).toMatchSnapshot();
    });
});
