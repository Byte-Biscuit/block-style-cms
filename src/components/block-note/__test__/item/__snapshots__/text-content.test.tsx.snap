// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormattedText > renders text and applies style classes from styles object 1`] = `
<DocumentFragment>
  <span
    class="font-bold italic underline text-red-600 bg-sky-100"
  >
    试文
  </span>
</DocumentFragment>
`;
